custom_model_runner/datarobot_drum/drum/gpu_predictors/triton_predictor.py

```diff
@@ -95,8 +98,29 @@ def has_read_input_data_hook(self):
     def _predict(self, **kwargs) -> RawPredictResponse:
         raise DrumCommonException("Predict structured is not supported")
 
+    def _get_inference_header_size(self, data):
+        try:
+            first_line = io.BytesIO(data).readline()
+            json.loads(first_line)  # read first line to check it's a valid json string
+            return len(first_line)
```

Comment:
What's in this JSON payload, can there be Unicode characters? I guess it doesn't matter if it does or not? Also does `first_line` contain the `\n` character?

Reply:
> What's in this JSON payload
example payload:
```
POST /v2/models/mymodel/infer HTTP/1.1
Host: localhost:8000
Content-Type: application/octet-stream
Inference-Header-Content-Length: <xx>
Content-Length: <xx+19>
{
  "model_name" : "mymodel",
  "inputs" : [
    {
      "name" : "input0",
      "shape" : [ 2, 2 ],
      "datatype" : "UINT32",
      "parameters" : {
        "binary_data_size" : 16
      }
    },
    {
      "name" : "input1",
      "shape" : [ 3 ],
      "datatype" : "BOOL",
      "parameters" : {
        "binary_data_size" : 3
      }
    }
  ],
  "outputs" : [
    {
      "name" : "output0",
      "parameters" : {
        "binary_data" : true
      }
    }
  ]
}
<16 bytes of data for input0 tensor>
<3 bytes of data for input1 tensor>
```
https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/protocol/extension_binary_data.html#examples

Reply:
> can there be Unicode characters

yes, unicode characters can be in the name of input and output layers.

Reply:
> Also does first_line contain the \n character

yes, the `first_line` must contain the `\n` character in order we can parse it. I added the example client code to the https://github.com/datarobot/datarobot-user-models/pull/966/files